feg/gateway/gtp/gtp_client.go

```diff
@@ -125,34 +141,43 @@ func (c *Client) Run(ctx context.Context) error {
 			return
 		}
 	}()
+	//TODO: remove this wait once there is a way to check when the listener is ready
 	return nil
 }
 
-func (c *Client) GetServerAddress() *net.UDPAddr {
-	return c.remoteAddr
-}
-
-func (c *Client) GetLocalAddress() *net.UDPAddr {
-	return c.localAddr
-}
-
 // Get preferred outbound ip of this machine
-func GetOutboundIP(testIp *net.UDPAddr) (net.IP, error) {
+func GetLocalOutboundIP(testIp *net.UDPAddr) (net.IP, error) {
 	connection, err := net.Dial("udp", testIp.String())
 	if err != nil {
 		return nil, err
 	}
+
 	defer connection.Close()
 	localAddr := connection.LocalAddr().(*net.UDPAddr)
 	return localAddr.IP, nil
 }
 
+// configOrDefaultTimeout sets a default timeout if config timeout is 0
+func configOrDefaultTimeout(configTimeout time.Duration) time.Duration {
+	if configTimeout == 0 {
+		return DefaultGtpTimeout
+	}
+	return configTimeout
+}
+
 //TODO: remove this once we find a way to safely wait for initialization of the service
 // WaitUntilClientIsReady is a hack to know when the client is ready and avoid null pointer issues using
 // the GTP-C client too early. Since go-gtp doesn't offer any visibuility on the readines of the connection
 // we use LocalAddrs as indicator
 func (c *Client) WaitUntilClientIsReady(count int) {
+
+	// TODO: only use those 3 waits for debugging
+	time.Sleep(time.Millisecond * 20)
+	time.Sleep(time.Millisecond * 20)
+	time.Sleep(time.Millisecond * 20)
+
```

Comment:
This is going to stay here for now. Otherwise test may fail on ci

There is a PR on go-gtp already approved to fix this issue. Still waiting for the new tag to apply the new version to our s8_proxy